modules/framework/jvm/src/main/scala/RunnerCompat.scala

```diff
@@ -57,9 +58,9 @@ trait RunnerCompat[F[_]] { self: sbt.testing.Runner =>
     val stillRunning             = new AtomicInteger(0)
     val waitForResourcesShutdown = new java.util.concurrent.Semaphore(0)
 
-    val tasksAndSuites = taskDefs.toList.map { taskDef =>
-      taskDef -> suiteLoader(taskDef)
-    }.collect { case (taskDef, Some(suite)) => (taskDef, suite) }
+    val tasksAndSuites = (taskDefs.toList.mapFilter { taskDef =>
+      suiteLoader(taskDef).map(loader => (taskDef, loader))
+    })
```

Comment:
The previous code flags up an unreachable case warning in Scala 3, although I'm not sure why:
```
[warn] 63 |    }.collect { case (taskDef, Some(suite)) => (taskDef, suite) }
[warn]    |                     ^^^^^^^^^^^^^^^^^^^^^^
[warn]    |                     Unreachable case
```

Reply:
Til about mapFilter !